crates/rustc_codegen_spirv/src/spirv_type_constraints.rs

```diff
@@ -372,7 +372,8 @@ pub fn instruction_signatures(op: Op) -> Option<&'static [InstSig<'static>]> {
         Op::VectorExtractDynamic => sig! { (Vector(T), _) -> T },
         Op::VectorInsertDynamic => sig! {
             // FIXME(eddyb) missing equality constraint between input and output vectors.
-            (Vector(T), T, _) -> Vector(T)
+            // (Vector(T), T, _) -> Vector(T)
+            (T, _, _) -> T
```

Comment:
LGTM, mind changing the comment to something like this? just so it's super clear what's going on:
```rust
            // FIXME(eddyb) was `(Vector(T), T, _) -> Vector(T)` but that was
            // missing an equality constraint between input and output vectors;
            // we should use `(Vector(T, N), T, _) -> Vector(T, N)`, or constrain
            // input and output vectors to have the same type some other way.
            (T, _, _) -> T
```